File: src/datenformat/ChatMessage.ts
export interface ChatMessage {
  id: string;
  conversationId: string;
  senderName: string;
  senderAvatarUrl?: string;
  content: string;
  timestamp: string; // ISO date string
  read: boolean;
  fileUrl?: string;
  fileName?: string;
  fileType?: string;
  fileSize?: number;
  duration?: number; // Dauer in Sekunden (für Sprachnachrichten)
  // Weitergeleiteter Beitrag
  forwardedBeitragId?: string;
  forwardedBeitragTitel?: string;
  forwardedBeitragLink?: string;
  forwardedBeitragAutor?: string;
}

export interface Conversation {
  id: string;
  participants: ConversationParticipant[];
  lastMessage?: ChatMessage;
  unreadCount: number;
  updatedAt: string; // ISO date string
}

export interface ConversationParticipant {
  name: string;
  avatarUrl?: string;
}

File: backend/src/main/java/eu/strietwald/social/backend/ChatMessage.java
package eu.strietwald.social.backend;

import java.time.Instant;

import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;
import org.springframework.data.mongodb.core.mapping.FieldType;
import org.springframework.data.mongodb.core.mapping.MongoId;

import lombok.Getter;
import lombok.Setter;

@Document(collection = "chat_messages")
@Getter
@Setter
public class ChatMessage {

    @MongoId(FieldType.OBJECT_ID)
    private String id;

    @Field
    @Indexed
    private String conversationId;

    @Field
    private String senderName;

    @Field
    private String senderAvatarUrl;

    @Field
    private String content;

    @Field
    @Indexed
    private Instant timestamp;

    @Field
    private boolean read;

    @Field
    private String fileUrl;

    @Field
    private String fileName;

    @Field
    private String fileType;

    @Field
    private Long fileSize;

    @Field
    private Long duration; // Dauer in Sekunden (für Sprachnachrichten)
}
